ledger/transaction.ts

```diff
@@ -34,15 +35,15 @@ export async function getTransactionData(
   ordinalUtxo?: UTXO,
 ) {
   // Get sender address unspent outputs
-  const unspentOutputs: UTXO[] = await esploraProvider.getUnspentUtxos(senderAddress);
-
-  let filteredUnspentOutputs = unspentOutputs;
-
-  if (ordinalUtxo) {
-    filteredUnspentOutputs = filterUtxos(unspentOutputs, [ordinalUtxo]);
-  }
-
-  const ordinalUtxoInPaymentAddress = filteredUnspentOutputs.length < unspentOutputs.length;
+  const [unspentOutputs, addressOrdinalsUtxos] = await Promise.all([
+    esploraProvider.getUnspentUtxos(senderAddress),
+    getOrdinalsUtxos(esploraProvider, network, senderAddress),
+  ]);
+
+  const ordinalUtxoInPaymentAddress = ordinalUtxo
+    ? unspentOutputs.some((utxo) => utxo.txid === ordinalUtxo.txid && utxo.vout === ordinalUtxo.vout)
+    : false;
+  const filteredUnspentOutputs = filterUtxos(unspentOutputs, addressOrdinalsUtxos);
```

Comment:
does this belong as a method on the esploraProvider class? or is this anyway refactored in consolidated transaction logic?

Reply:
Yeah, it is refactored. We can't really put it in the esplora api provider either because that is exclusively for esplora API endpoints as it's the URL that the user can set in the  app. That provider is best kept clean.

Reply:
oh right. this also relies on a xord endpoint